src/Castle.Core/Components.DictionaryAdapter/DictionaryAdapterFactory.cs

```diff
@@ -440,22 +468,21 @@ private static void PreparePropertyMethod(PropertyDescriptor descriptor, ILGener
 						.AddBehaviors(typeBehaviors.OfType<IDictionaryInitializer>    ().Cast<IDictionaryBehavior>());
 #endif
 
-			CollectProperties(type, property =>
+			CollectProperties(type, (property, reflectedType) =>

```

Comment:
`PropertyInfo.ReflectedType` is gone from .Net Core.  The recommended change is to use the reference to the object being reflected to get this value.